pkg/controllers/provisioningv2/rke2/common.go

```diff
@@ -206,3 +213,17 @@ func CopyMapWithExcludes(destination map[string]string, source map[string]string
 		}
 	}
 }
+
+func ClientSetFromCache(namespace, clusterName string, rancherClusterCache ranchercontrollers.ClusterCache, kubeconfigManager *kubeconfig.Manager) (*kubernetes.Clientset, error) {
```

Comment:
I believe this code is also used in `machinenodelookup`. Can we use this function there too?